pkg/runner/cluster_k8s.go

```diff
@@ -303,54 +305,114 @@ func (*ClusterK8sRunner) CompatibleBuilders() []string {
 	return []string{"docker:go"}
 }
 
-func (*ClusterK8sRunner) CollectOutputs(ctx context.Context, input *api.CollectionInput, w io.Writer) error {
+func (c *ClusterK8sRunner) initRunner(cfg ClusterK8sRunnerConfig) {
+	// init Kubernetes runner
+	once.Do(func() {
+		c.config = defaultKubernetesConfig()
+
+		var err error
+		workers := 20
+		c.pool, err = newPool(workers, c.config)
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		c.podResourceCPU = resource.MustParse(cfg.PodResourceCPU)
+		c.podResourceMemory = resource.MustParse(cfg.PodResourceMemory)
+	})
+}
+
+func (c *ClusterK8sRunner) CollectOutputs(ctx context.Context, input *api.CollectionInput, w io.Writer) error {
 	log := logging.S().With("runner", "cluster:k8s", "run_id", input.RunID)
+
+	//TODO: we shouldn't have to init the runner in every handler
 	cfg := *input.RunnerConfig.(*ClusterK8sRunnerConfig)
+	c.initRunner(cfg)
+
+	client := c.pool.Acquire()
+	defer c.pool.Release(client)
 
 	log.Info("collecting outputs")
+	err := c.ensureCollectOutputsPod(ctx)
+	if err != nil {
+		return err
+	}
 
-	sess, err := session.NewSession(&aws.Config{
-		Region: aws.String(cfg.OutputsBucketRegion)},
-	)
+	file := fmt.Sprintf("/outputs/%s.tgz", input.RunID)
+	args := fmt.Sprintf("kubectl exec %s cat %s -- ", collectOutputsPodName, file)
```

Comment:
I considered doing this through the Kubernetes API, but couldn't get it to work, and the code was much more complex, compared to just calling `kubectl` with `exec`.